components/annot-tool-group.vue

```diff
@@ -0,0 +1,323 @@
+                            <template #cell(missing_value)="row">
+                                <b-button
+                                    variant="danger"
+                                    @click="declareMissing(row.item)">
+                                    {{ uiText.missingValueButton }}
+                                </b-button>
+                            </template>
+                        </b-table>
+
+                    </b-tab>
+
+                </b-tabs>
+
+            </b-card-body>
+
+        </b-card>
+
+        <!-- Button to save the annotated data of all group items to the store -->
+        <b-row>
+            <b-button
+                variant="success"
+                @click="applyAnnotation()">
+                {{ uiText.saveButton }}
+            </b-button>
+        </b-row>
+    </div>
+
+</template>
+
+<script>
+
+    // Allows for reference to store data by creating simple, implicit getters
+    import { mapGetters } from "vuex";
+
+    export default {
+
+        props: {
+
+            details: { default: () => {}, required: true, type: Object },
+            idField: { default: () => {}, required: true, type: String },
+            relevantColumns: { default: () => [], required: true, type: Array },
+            uniqueValues: { default: () => {}, required: true, type: Object },
+            uniqueValuesToSubjectMap: { default: () => {}, required: true, type: Object }
+        },
+
+        inject: [
+
+            "dataTable",
+            "isMissingValue",
+            "valueDescription"
+        ],
+
+        name: "AnnotToolGroup",
+
+        data() {
+
+            return {
+
+                availableSubjects: 0,
+                totalSubjects: 0,
+
+                fields: [
+
+                    "column_name",
+                    "raw_value",
+                    "description",
+                    "missing_value"
+                ],
+
+                tableItems: {},
+
+                uiText: {
+
+                    availability: "Subjects available for this tool group: ",
+                    generalInstructions: "Indicate which values are 'missing' for each column in the group",
+                    missingValueButton: "Mark as missing",
+                    saveButton: "Save Annotation"
+                },
+
+                subjectAvailability: {}
+            };
+        },
+
+        computed: {
+
+            ...mapGetters([
+
+                "getColumnOfCategory"
+            ])
+        },
+
+        watch: {
+
+            "$store.state.missingColumnValues": {
+
+                deep: true,
+                handler(p_newValue, p_oldValue) {
+
+                    // When missing column values are updated in the store,
+                    // load table data from 'uniqueValues' and update 'subjectAvailability' if values for it are not missing
+                    this.refreshTableItems();
+                }
+            },
+
+            subjectAvailability: {
+
```

Comment:
Just commenting here for the future: there is some pretty fancy magic going on to keep this guy in sync with changes outside of the component (indirect listening to changes in the missingValue store variable). Because these links are largely implicit (listenting to a local variable because it can be changed by another listener calling a method, and so on) I believe this will become hard to remember right quick, so it might be a good candidate to have another look at in the future and maybe move it to the store.